src/endpoints/translate.js

```diff
@@ -61,39 +64,88 @@ router.post('/libre', jsonParser, async (request, response) => {
     }
 });
 
+
 router.post('/google', jsonParser, async (request, response) => {
     try {
         const { generateRequestUrl, normaliseResponse } = require('google-translate-api-browser');
-        const text = request.body.text;
+        let text = request.body.text;
         const lang = request.body.lang;
 
         if (!text || !lang) {
             return response.sendStatus(400);
         }
 
-        console.log('Input text: ' + text);
+        console.log('Input text:', text);
+        //console.log('----------');
+
+        // Apply formatting only if the translation language is Russian
+        if (lang === 'ru') {
+             // Replace quotation marks and asterisks with special markers before translating
+            const openQuote = '__OPEN_QUOTE__ ';
+            const closeQuote = ' __CLOSE_QUOTE__';
+            const openStar = '__OPEN_STAR__ ';
+            const closeStar = ' __CLOSE_STAR__';
+             // A very rare case
+            const very_rare_execption = '__OPEN_STAR__, ';
+
+            // Use counters to alternate between opening and closing markers
+            let quoteCounter = 0;
+            let starCounter = 0;
+
+            text = text.replace(/"/g, () => (quoteCounter++ % 2 === 0 ? openQuote : closeQuote));
+            text = text.replace(/\*/g, () => (starCounter++ % 2 === 0 ? openStar : closeStar));
+
+            //console.log('Pre-Input text:', text);
+            //console.log('----------');
+        }
 
         const url = generateRequestUrl(text, { to: lang });
 
         https.get(url, (resp) => {
-            let data = '';
+            let data = [];
 
             resp.on('data', (chunk) => {
-                data += chunk;
+                data.push(chunk);
             });
 
             resp.on('end', () => {
                 try {
-                    const result = normaliseResponse(JSON.parse(data));
-                    console.log('Translated text: ' + result.text);
-                    return response.send(result.text);
+                    let result;
+                    if (lang === 'ru') {
+                        // Decoding for Russian language
```

Comment:
This should be applied to other languages as well. I've seen this being reported